nemo/export/onnx_llm_exporter.py

```
@@ -0,0 +1,512 @@
+
+    Example:
+        from nemo.export.onnx_llm_exporter import OnnxLLMExporter
+
+        onnx_llm_exporter = OnnxLLMExporter(
+            onnx_model_dir="/path/for/onnx_model/files",
+            model_name_or_path="/path/for/model/files",
+        )
+
+        onnx_llm_exporter.export(
+            input_names=["input_ids", "attention_mask", "dimensions"],
+            output_names=["embeddings"],
+        )
+
+        output = trt_llm_exporter.forward(["Hi, how are you?", "I am good, thanks, how about you?"])
+        print("output: ", output)
+    """
+
+    def __init__(
+        self,
+        onnx_model_dir: str,
+        model: Optional[torch.nn.Module] = None,
+        tokenizer=None,
+        model_name_or_path: str = None,
+        load_runtime: bool = True,
+    ):
+        """
+        Initializes the ONNX Exporter.
+
+        Args:
+            onnx_model_dir (str): path for storing the ONNX model files.
+            model (Optional[torch.nn.Module]): torch model.
+            tokenizer (HF or NeMo tokenizer): tokenizer class.
+            model_name_or_path (str): a path for ckpt or HF model ID
+            load_runtime (bool): load ONNX runtime if there is any exported model available in
+                                 the onnx_model_dir folder.
+        """
+        self.onnx_model_dir = onnx_model_dir
+        self.model_name_or_path = model_name_or_path
+        self.onnx_model_path = str(Path(onnx_model_dir) / "model.onnx")
+        self.model = model
+        self.tokenizer = tokenizer
+        self.model_input_names = None
+        self.model_output_names = None
+        self.onnx_runtime_session = None
+        self.calibration_data = None
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.quant_max_batch_size = None
+
+        if self.model_name_or_path is not None:
+            if model is not None:
+                warnings.warn("A model was also passed but it will be overridden.")
```

Should we raise an error instead to remove ambiguity?